core/market/decentralized/src/rest_api/requestor.rs

```diff
@@ -135,11 +135,15 @@ async fn create_agreement(
 
 #[actix_web::get("/agreements/{agreement_id}")]
 async fn get_agreement(
-    _market: Data<Arc<MarketService>>,
-    _path: Path<PathAgreement>,
-    _id: Identity,
-) -> HttpResponse {
-    HttpResponse::NotImplemented().finish()
+    market: Data<Arc<MarketService>>,
+    body: Path<PathAgreement>,
+    id: Identity,
+) -> impl Responder {
```

Comment:
This is an exact copy & paste of providers counterpart. Is it possible to reuse this code?